### PR TITLE
Add ShellCheck

### PR DESCRIPTION
**Pull request by zSALLAZAR**
## Introduction
This PR adds [ShellCheck](https://github.com/koalaman/shellcheck) to the CI

## Follow-up

Fix the checks that are probably going to fail

## Tests

I tested this PR by doing the following (tick all that apply):
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [ ] Playtesting using a Minecraft client (provide screenshots or a video)
- [ ] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

### CODE REVIEW

**Comment by SOF3** [comment]
Is travis.sh still used?

**Comment by dktapps** [comment]
Yep, it's used for integration tests: https://github.com/pmmp/PocketMine-MP/blob/df4ada81e5d74a14046f27cf44a37dcee69d657e/.github/workflows/main-php-matrix.yml#L119

I just never bothered to rename it when we shifted to GitHub Actions

**Comment by ShockedPlot7560** [comment]
Your CI is failing

**Comment by zSALLAZAR** [comment]
> Your CI is failing

Fixed

**Comment by IvanCraft623** [comment]
Looks OK! Thanks for your contribution!

**Review by IvanCraft623 — APPROVED** [review]

File: tests/travis.sh [hunk 1]
```diff
@@ -45,7 +48,7 @@ if [ "$result" != "" ]; then
 	echo "$result"
 	echo Some tests did not complete successfully, changing build status to failed
 	exit 1
-elif [ $(grep -c "ERROR\|CRITICAL\|EMERGENCY" "$DATA_DIR/server.log") -ne 0 ]; then
+elif [ "$(grep -c "ERROR\|CRITICAL\|EMERGENCY" "$DATA_DIR/server.log")" -ne 0 ]; then
```
Comment (dktapps) [inline tests/travis.sh:51]: Do we not need to escape the inner quotes here?

Comment (ShockedPlot7560) [reply tests/travis.sh:51]: Maybe the CI failling if not ? I wouldn't know why else

Comment (dktapps) [reply tests/travis.sh:51]: Yeah you're right. The PR is its own proof of validity I guess.

Comment (dktapps) [reply tests/travis.sh:51]: Looks like there's no escaping needed according to this: https://unix.stackexchange.com/questions/118433/quoting-within-command-substitution-in-bash

**Review by dktapps — APPROVED** [review]

**Pull request merged**